### ISSUE TITLE
Instances ignore the names they're assigned in project.json

### ISSUE THREAD

**Issue by icanismar** [Iron-Stag-Games/Lync#23]
For instances that are assigned a `$path` in the default.project.json file, upon modifying them, they seem to default to their file name, which causes them to duplicate, and sync to the wrong instance. (apologies if this sounds confusing, hopefully the reproduction steps clear things up)

**To Reproduce:**

1. Have a project.json file that has an Instance with a `$path` property

```json
{
  "name": "Ballistic",
  "tree": {
    "ReplicatedFirst": {
      "$className": "ReplicatedFirst",
      "ClientLoader": {
        "$path": "source/ballistic/game.client.luau"
      }
    }
  }
}
```

2. Begin syncing, you should observe that initially the instance gets created with the proper name "ClientLoader", as also backed up by the logs

```
Mapping Instance 'tree/ReplicatedFirst/ClientLoader' -> [source\ballistic\game.client.luau]
Mapping Lua 'tree/ReplicatedFirst/ClientLoader' -> [source\ballistic\game.client.luau]
```

3. Modify the file and notice how another instance is created called "game"

```
M [source\ballistic\game.client.luau]
Mapping Lua 'tree/ReplicatedFirst/game' -> [source\ballistic\game.client.luau]
```
![image](https://github.com/Iron-Stag-Games/Lync/assets/37914143/d41189e4-1bdc-4ae7-99fb-cd2e845877a5)

4. From this point any further modifications made to `source\ballistic\game.client.luau` will apply to `ReplicatedFirst/game`, rather than the expected `ReplicatedFirst/ClientLoader`

**Note:** I have only tested this with luau files, although I assume the same issue would persist with other formats as well.

**Comment by hoontee** [comment]
It's amazing this wasn't caught earlier. I'll get this fixed.

**Comment by hoontee** [comment]
Fixed! Please update to [Alpha 15.4+](https://github.com/Iron-Stag-Games/Lync/releases) or [enable automatic updates](https://github.com/Iron-Stag-Games/Lync#configuration).
